python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -750,17 +776,35 @@ def _throughput_calculate_helper(iterrun, baseline_time, func, *args):
     return np.mean(time_list) * 1000, True
 
 
+def _signature_check(function):
+    '''
+    A quick helper to judge whether input function is following this calling
+    method `metric(pred, target)`.
+    '''
+    import inspect
+    sig = inspect.signature(function)
+    if len(sig.parameters.values()) != 2:
+        return False
+    param1_name = list(sig.parameters.values())[0].name
+    param2_name = list(sig.parameters.values())[0].name
+    if "pred" in param1_name or "pred" in param2_name:
+        return True
+    if "target" in param2_name or "target" in param2_name:
+        return True
```

Comment:
does this make sense?

Reply:
here we assume user's custom metric function is like this ` function_name(pred=xxx, target=xxx)`, a example is : https://github.com/intel-analytics/BigDL/blob/main/python/nano/example/pytorch/inference_optimizer/resnet/inference_pipeline.py

I think we have to limit the function signature here, otherwise we can't distinguish metric(pred, target) from metric (model, data_loader).